Maker-Agents-Suit-Module/OSMAgent/src/TimeTracking.ts

```diff
@@ -0,0 +1,35 @@
+export default class TimeTracking {
+  hourStatus = false; // indicates if the function got called
+  hour = 0; // keeps track of the hour
+
+  // make sure hour gets updates and consequently status
+  initialUpdate(timestamp: number): void {
+    if (this.hour !== this.getHour(timestamp)) {
+      this.hour = this.getHour(timestamp);
+      this.hourStatus = false;
+    }
+  }
+
+  getHour(timestamp: number): number {
+    const nd = new Date(timestamp * 1000);
+    return nd.getHours();
+  }
+
+  getMinute(timestamp: number): number {
+    var d = new Date(timestamp * 1000); //x1000 to convert from seconds to milliseconds var s = d.toUTCString() s = s.substring(0,s.indexOf("GMT")) + "UTC" //change the confusing 'GMT' to 'UTC'
+    return d.getMinutes();
+  }
+
+  getTime(timestamp: number): boolean {
```

Comment:
I would rather use a name more explicit for this, something like `isInFirstTenMins` or `isCorrectTime`, using `getTime` I had to come to look the code for having an idea about what is doing that function.